slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/dao/SignedBlocksDao.java

```diff
@@ -12,21 +12,36 @@
  */
 package tech.pegasys.web3signer.slashingprotection.dao;
 
+import java.util.List;
 import java.util.Optional;
 import java.util.stream.Stream;
 
+import org.apache.tuweni.bytes.Bytes;
 import org.apache.tuweni.units.bigints.UInt64;
 import org.jdbi.v3.core.Handle;
 
 public class SignedBlocksDao {
 
-  public Optional<SignedBlock> findExistingBlock(
-      final Handle handle, final int validatorId, final UInt64 slot) {
+  public List<SignedBlock> findBlockForSlotWithDifferentSigningRoot(
```

Comment:
Looks like we don't need a list do just need to know if there is a block that exists? So could add a limit 1 and return Optional<Block>

Reply:
Yeah, I was wondering, but then decided that using a list is more generic, so potentially has other uses - no point in limiting the flexibility of this class.